src/PIL/ImageDraw.py

```diff
@@ -110,7 +110,7 @@ def _getink(self, ink, fill=None):
                 if self.palette and not isinstance(ink, numbers.Number):
                     ink = self.palette.getcolor(ink)
                 ink = self.draw.draw_ink(ink, self.mode)
-            if fill is not None:
+            if fill is not None and ink != fill:
```

Comment:
Fill and ink are variable type values, e.g. 'white' and (255,255,255) are equivalent, but not equal. You can only really compare them for equality after the ImageColor.getcolor call.

Reply:
Okay, thanks. I've removed that line change.